client/state/stats/lists/selectors.js

```diff
@@ -134,20 +122,11 @@ export const getSiteStatsMaxPostsByDay = createSelector(
  * @param  {Object}  query    Stats query object
  * @return {?Number}          Max number of posts by day
  */
-export const getSiteStatsTotalPostsForStreakQuery = createSelector(
-	( state, siteId, query ) => {
-		return reduce(
-			getSiteStatsPostStreakData( state, siteId, query ),
-			( posts, sum ) => {
-				return sum + posts;
-			},
-			0
-		);
-	},
-	( state, siteId, query ) => getSiteStatsForQuery( state, siteId, 'statsStreak', query ),
-	( state, siteId, query ) => {
-		const serializedQuery = getSerializedStatsQuery( query );
-		return [ siteId, 'statsStreakMax', serializedQuery ].join();
+export const getSiteStatsTotalPostsForStreakQuery = treeSelect(
+	( state, siteId, query ) => [ getSiteStatsPostStreakData( state, siteId, query ) ],
+	( [ postStreakData ] ) => reduce( postStreakData, ( sum, posts ) => sum + posts, 0 ),
```

Comment:
we could even use the [sum](https://lodash.com/docs/4.17.4#sum) function here

Reply:
I'm going to remove the "total posts" and "max posts" selectors in a follow-up PR, as they are not really needed. I just didn't want to make this PR more complicated. And I already finished their `treeSelect` implementation when I realized they can be removed :(